analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Pattern.cs

```diff
@@ -50,9 +50,10 @@ private static SymbolicConstraint LearnBranchingConstraint(ProgramState state, I
         {
```

Comment:
Can we be more explicit, that this is about `ObjectContraint`s only? Preferable by replacing `SymbolContraint` with `ObjectConstraint` or by replacing "Constraint" with "ObjectConstriant" in the methods.

If we want to learn more than just ObjectConstraint, we need to return `SymbolConstraint[]` at one point anyway. But I don't know if we should mix different constraints (separation of concerns).

Reply:
We can not, `ConstraintFromConstantPattern` learns `BoolConstraint` too. And we'll be learning other constraints int he future